extension/observer/ecstaskobserver/factory_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecstaskobserver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+func TestFactoryCreatedExtensionIsObservable(t *testing.T) {
+	etoFactory := NewFactory()
+	eto, err := etoFactory.CreateExtension(
+		context.Background(), componenttest.NewNopExtensionCreateSettings(), etoFactory.CreateDefaultConfig(),
+	)
+	require.NoError(t, err)
+	require.NotNil(t, eto)
+	asObservable, ok := eto.(observer.Observable)
```

Comment:
I suggest using https://pkg.go.dev/github.com/stretchr/testify/assert#Implements to check if it implements the interface.

However, this feels like a redundant tests considering there is a compile time check

Reply:
This test is for the factory method which wouldn't enforce the returned helper type at compile time, though the extension itself does have an added implementation check that could make this redundant.  Given the level of indirection I think it's a helpful assurance.

Updated to use the implementation helper.